### PR TITLE
More Changes

### PR DESCRIPTION
**Pull request by singhdivya-2895**
1. Post Route
2. async Await
3. JsonAttribute

### CODE REVIEW

File: CmswebApI/DTOs/CourseDto.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+namespace CmswebApI.DTOs
+{
+    public class CourseDto
+    {
+        public int CourseID { get; set; }
+        public string CourseName { get; set; }
+        public int CourseDuration { get; set; }
+        public string Source { get; } = "I am from Dto";
+        
+         [JsonConverter(typeof(JsonStringEnumConverter))]
```
Comment (Pankaj-chaudhary) [inline CmswebApI/DTOs/CourseDto.cs:16]: Why have you done this?

File: CmswebApI/Startup.cs [hunk 2]
```diff
@@ -27,6 +27,14 @@ public Startup(IConfiguration configuration)
...
         {
+            // AddSingleton: This lifetime creates a single instance of the service that is shared across the entire application. 
+            // Once the service is created, it is reused for each subsequent request for that service.
+
+            // AddTransient: This lifetime creates a new instance of the service for each request. 
+            // This is useful for services that have a short lifespan and are not thread-safe.
+
+            // AddScoped: This lifetime creates a new instance of the service for each HTTP request. 
+            // The instance is shared within the scope of the request, so it can be used by multiple components during the request.
             services.AddSingleton<ICmsrepository, InMemoryCmsRepository>();
```
Comment (Pankaj-chaudhary) [inline CmswebApI/Startup.cs:38]: Good notes

File: CmswebApI/DTOs/CourseDto.cs (hunk 1 above)
Comment (singhdivya-2895) [reply CmswebApI/DTOs/CourseDto.cs:16]: Because, it was expecting and returning an integer value for Enum by default.

**Review by Pankaj-chaudhary — APPROVED** [review]
Good work on gitignore.

**Pull request merged**